backend/battle/tests/test_view.py

```diff
@@ -74,3 +81,178 @@ def test_verify_if_not_returns_winner_missing_pokemons_teams(self):
+
+        self.assertCountEqual(battles_initial, response_qs_initial)
+
+        baker.make('battle.Battle', creator=self.user, _quantity=90)
+        battles_updated = Battle.objects.filter(creator=self.user)
+        response_updated = self.client.get(reverse('battle_list'))
+        response_qs_updated = response_updated.context_data.get('battle_list')
+
+        self.assertCountEqual(battles_updated, response_qs_updated)
+
+
+class BattleCreateViewTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.client = Client()
+        self.creator = User.objects.create(email='example@example.com')
+        self.creator.set_password('admin')
+        self.creator.save()
+        self.opponent = baker.make('users.User')
+
+    def test_create_battle_successfully(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        self.client.post(reverse('battle'), battle_data)
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+        self.assertTrue(battle)
+
+    def test_if_returns_error_when_creator_and_opponent_are_the_same_user(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.creator.email,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        response = self.client.post(reverse('battle'), battle_data)
+        self.assertEqual(
+            response.context_data['form'].errors['__all__'][0],
+            "ERROR: You can't challenge yourself.")
+
+    def test_if_returns_error_when_opponent_doesnt_exist(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": 100,
+        }
+        self.client.login(username=self.creator.email, password='admin')
+        with self.assertRaises(KeyError):
+            self.client.post(reverse('battle'), battle_data)
+
+    @patch("battle.battles.email.send_templated_mail")
+    def test_if_battle_invitation_email_is_sent(self, mock_templated_mail):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+
+        self.client.login(username=self.creator.email, password='admin')
+
+        self.client.post(reverse("battle"), battle_data)
+
+        battle = Battle.objects.filter(
+            creator=self.creator, opponent=self.opponent)
+
+        self.assertTrue(battle)
+
+        mock_templated_mail.assert_called_with(
+            template_name="invite_challenge",
+            from_email=settings.FROM_EMAIL,
+            recipient_list=[self.opponent.email],
+            context={
+                'creator': self.creator.email,
+            },
+        )
+
+    @patch('battle.tasks.run_battle')
+    def test_if_run_battle_task(self, result_battle_mock):
```

Comment:
I think this test is misplaced. You should test if the `run_battle_and_send_result_email` is called properly and then test if the `run_battle_and_send_result_email` separately

Reply:
You can remove this test since it doesn't make sense to be on `BattleCreateViewTest` and it's still untouched